app/code/BodenkoVV/AskQuestion/Controller/Adminhtml/Question/PostDataProcessor.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace BodenkoVV\AskQuestion\Controller\Adminhtml\Question;
+
+use BodenkoVV\AskQuestion\Model\DomValidationState;
+use Magento\Framework\App\Arguments\ValidationState;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Config\Dom\ValidationException;
+use Magento\Framework\Config\Dom\ValidationSchemaException;
+
+/**
+ * Class PostDataProcessor
+ * @package Magento\Cms\Controller\Adminhtml\Page
+ */
+class PostDataProcessor
+{
+    /**
+     * @var \Magento\Framework\Stdlib\DateTime\Filter\Date
+     */
+    protected $dateFilter;
+
+    /**
+     * @var \Magento\Framework\View\Model\Layout\Update\ValidatorFactory
+     */
+    protected $validatorFactory;
+
+    /**
+     * @var \Magento\Framework\Message\ManagerInterface
+     */
+    protected $messageManager;
+
+    /**
+     * @var DomValidationState
+     */
+    private $validationState;
+
+    /**
+     * @param \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter
+     * @param \Magento\Framework\Message\ManagerInterface $messageManager
+     * @param \Magento\Framework\View\Model\Layout\Update\ValidatorFactory $validatorFactory
+     * @param DomValidationState $validationState
+     */
+    public function __construct(
+        \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter,
+        \Magento\Framework\Message\ManagerInterface $messageManager,
+        \Magento\Framework\View\Model\Layout\Update\ValidatorFactory $validatorFactory,
+        DomValidationState $validationState = null
+    ) {
+        $this->dateFilter = $dateFilter;
+        $this->messageManager = $messageManager;
+        $this->validatorFactory = $validatorFactory;
+        /** @var ValidationState $this */
+        $this->validationState = $validationState
+            ?: ObjectManager::getInstance()->get(DomValidationState::class);
+    }
+
+    /**
+     * Filtering posted data. Converting localized data if needed
+     *
+     * @param array $data
+     * @return array
+     */
+    public function filter($data)
+    {
+        $filterRules = [];
+
+        foreach (['custom_theme_from', 'custom_theme_to'] as $dateField) {
+            if (!empty($data[$dateField])) {
+                $filterRules[$dateField] = $this->dateFilter;
+            }
+        }
+
+        return (new \Zend_Filter_Input($filterRules, [], $data))->getUnescaped();
+    }
+
+    /**
+     * Validate post data
+     *
+     * @param array $data
+     * @return bool     Return FALSE if some item is invalid
+     */
+    public function validate($data)
+    {
+        if (!empty($data['layout_update_xml']) || !empty($data['custom_layout_update_xml'])) {
```

Comment:
This is for layout xml for CMS pages, don't need this for question editing. Or you really use some field for xml in your questions?

Reply:
+